jina/serve/runtimes/worker/http_sagemaker_app.py

```diff
@@ -115,6 +126,48 @@ async def post(body: input_model, response: Response):
             else:
                 return output_model(data=resp.docs, parameters=resp.parameters)
 
+        @app.api_route(**app_kwargs)
+        async def post(request: Request):
+            content_type = request.headers.get('content-type')
+            if content_type == 'application/json':
+                json_body = await request.json()
+                return await process(input_model(**json_body))
+
+            elif content_type in ('text/csv', 'application/csv'):
+                bytes_body = await request.body()
+                csv_body = bytes_body.decode('utf-8')
+                if not is_valid_csv(csv_body):
+                    raise HTTPException(
+                        status_code=400,
+                        detail='Invalid CSV input. Please check your input.',
+                    )
+
+                # NOTE: Sagemaker only supports csv files without header, so we enforce
+                # the header by getting the field names from the input model.
+                # This will also enforce the order of the fields in the csv file.
+                # This also means, all fields in the input model must be present in the
+                # csv file including the optional ones.
+                field_names = [f for f in input_doc_list_model.__fields__]
```

Comment:
I believe that `DocArray` provides a `from_csv` method

Reply:
Doesn't work without csv header. Sagemaker only supports csvs without header.

Reply:
can u open a ticket on DocArray with this?